contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -587,6 +591,15 @@ init_tsql_coerce_hash_tab(PG_FUNCTION_ARGS)
                                 PointerGetDatum(buildoidvector(&castsource, 1)),
                                 ObjectIdGetDatum(sys_nspoid));
                     if (!OidIsValid(entry->castfunc))
+                    {
+                        /* also search cast function with 3 input arguments */
+                        argTypes[0] = castsource;
+                        entry->castfunc = GetSysCacheOid3(PROCNAMEARGSNSP, Anum_pg_proc_oid,
+                                    CStringGetDatum(tsql_cast_raw_infos[i].castfunc),
+                                    PointerGetDatum(buildoidvector(argTypes, 3)),
+                                    ObjectIdGetDatum(sys_nspoid));
+                    }
+                    if (!OidIsValid(entry->castfunc))
```

Comment:
You can move this if-block into the above if-block because when `entry->castfunc` is valid this if-check is redundant.

Reply:
Done!